llpc/context/llpcCompiler.cpp

```diff
@@ -211,10 +211,28 @@ static void fatalErrorHandler(void *userData, const std::string &reason, bool ge
 #endif
 }
 
+static void InlineAsmDiagHandler(const SMDiagnostic &SMD, void *Context, unsigned LocCookie) {
+  bool *HasError = static_cast<bool *>(Context);
+  if (SMD.getKind() == SourceMgr::DK_Error)
+    *HasError = true;
+
+  SMD.print(nullptr, errs());
+
+  // For testing purposes, we print the LocCookie here.
+  if (LocCookie)
+    WithColor::note() << "!srcloc = " << LocCookie << "\n";
+}
+
 // =====================================================================================================================
 // Handler for diagnosis in pass run, derived from the standard one.
 class LlpcDiagnosticHandler : public DiagnosticHandler {
+  bool *HasError;
```

Comment:
Please add it to a private section with explicit "private". And a private section always follows a public section (not in the beginning of a class).
HasError -> m_hasError